state/protocol/badger/state.go

```diff
@@ -57,12 +61,10 @@ type State struct {
 	// Caution: A node that joined in a later epoch past the spork, the node will likely _not_
 	// know the spork's root block in full (though it will always know the height).
 	sporkRootBlockHeight uint64
-	// cachedLatestFinal is the *latest* finalized block header, which we can cache here,
+	// cachedLatest caches both the *latest* finalized header and sealed header,
 	// because the protocol state is solely responsible for updating it.
-	cachedLatestFinal *atomic.Pointer[cachedHeader]
-	// cachedLatestSealed is the *latest* sealed block headers, which we can cache here,
-	// because the protocol state is solely responsible for updating it.
-	cachedLatestSealed *atomic.Pointer[cachedHeader]
+	// finalized header and sealed header can be cached together since they are updated together atomically
```

Comment:
```suggestion
	// Finalized header and sealed header should be cached together for conceptual consistency
	// since they are also updated together atomically.
```

Reply:
Will apply in a separate PR